qiskit_ibm/runtime/ibm_runtime_service.py

```diff
@@ -273,10 +278,10 @@ def upload_program(
             max_execution_time: Optional[int] = None,
             description: Optional[str] = None,
             version: Optional[float] = None,
-            backend_requirements: Optional[str] = None,
-            parameters: Optional[List[ProgramParameter]] = None,
-            return_values: Optional[List[ProgramResult]] = None,
-            interim_results: Optional[List[ProgramResult]] = None
+            backend_requirements: Optional[Dict] = None,
+            parameters: Optional[Dict] = None,
+            return_values: Optional[Dict] = None,
+            interim_results: Optional[Dict] = None
```

Comment:
The original intend was to make it easier for users to define metadata programatically. For example, I could do 

```
param1 = ProgramParameter(name="foo", description="foo")
param2 =ProgramParameter(name="bar", description="bar") 
upload_program(parameters=[param1, param2])
```

If they now have to pass in entire JSON schemas, is there any benefit of having these arguments, which does nothing more than adding them to `spec`? Or should we just make users pass in `metadata`?

Reply:
I don't see any benefit. Yes, I think we should just have `data` and `metadata` as only parameters for `upload_program`, if that's what you mean?.